### ISSUE TITLE
Err value: invalid type: string "myfieldvalue", expected a borrowed string' after updating dependencies

### ISSUE THREAD

**Issue by lcmgh** [Pantonshire/enumscribe#3]
```rust
#[derive(EnumSerialize, EnumDeserialize, EnumIter, PartialEq, Eq, Clone, Copy, Debug, Display)]
#[enumscribe(case_insensitive)]
pub enum Hub {
    Emea,
    Us,
    Cn,
}
``` 

After upgrading from v0.2 to v0.3 I am getting this error: Err value: invalid type: string "myfieldvalue", expected a borrowed string'

**Comment by pantonshire** [comment]
I am unable to reproduce this issue. Could you provide a more complete example? Since the error appears to originate from `serde`, what deserializer are you using?

**Comment by pantonshire** [comment]
Ah, disregard the above; I've managed to reproduce.

The issue appears to be when the string value being deserialized needs to be transformed in some way by the deserializer e.g. if it contains an escape sequence like `\n` in JSON, which prevents it from being deserialized to a `&str`. This didn't use to be an issue because value was deserialized to a `String` in v0.2, but this was changed to a `&str` to support `no_std` in v0.3.

I'll have a go at fixing this this evening.
